go-controller/pkg/informer/informer.go

```diff
@@ -161,6 +161,10 @@ func (e *eventHandler) Run(threadiness int, stopCh <-chan struct{}) error {
 
 // enqueue adds an item to the workqueue
 func (e *eventHandler) enqueue(obj interface{}) {
+	// ignore objects that are already set for deletion
+	if obj.(metav1.Object).GetDeletionTimestamp() != nil {
```

Comment:
perhaps

`!obj.(metav1.Object).GetDeletionTimestamp().IsZero`?

Reply:
I've asked to the apimachinery folks about this, because the [namespace controller uses this format](https://github.com/kubernetes/kubernetes/blob/323f34858de18b862d43c40b2cced65ad8e24052/pkg/controller/namespace/namespace_controller.go#L127)

```
	if namespace.DeletionTimestamp == nil || namespace.DeletionTimestamp.IsZero() {
```

They told me checking for nil should be enough

Reply:
Well `IsZero()` method already checks if the receiver is `nil`.

```
// IsZero returns true if the value is nil or time is zero.
func (t *Time) IsZero() bool {
	if t == nil {
		return true
	}
	return t.Time.IsZero()
}
```

I think the `namespace controller` needs a fix :).

Reply:
didn't know it checks for nil, let's do the right way then